src/main/webapp/app/exercises/shared/result/updating-result.component.ts

```diff
@@ -149,4 +130,28 @@ export class UpdatingResultComponent implements OnChanges, OnDestroy {
         }
         return MissingResultInfo.FAILED_PROGRAMMING_SUBMISSION_ONLINE_IDE;
     }
+
+    private shouldUpdateSubmissionState(submission?: Submission): boolean {
+        // The updating result must ignore submissions that are ungraded if ungraded results should not be shown
+        // (otherwise the building animation will be shown even though not relevant).
+        return (
+            this.showUngradedResults ||
+            !submission ||
+            !this.exercise.dueDate ||
+            submission.type === SubmissionType.INSTRUCTOR ||
+            submission.type === SubmissionType.TEST ||
+            dayjs(submission.submissionDate).isBefore(getExerciseDueDate(this.exercise, this.participation))
```

Comment:
This condition was inverted before.

The remaining changes just refactor the code from inside the pipe to separate methods but are otherwise unchanged.